python/tests/unit/arcticdb/version_store/test_date_range.py

```diff
@@ -40,3 +42,80 @@ def test_arctic_date_range_max():
     assert result.end_ts == 9223372036854775807
     assert pd.Timestamp(result.start_ts, unit="ns") == pd.Timestamp.min
     assert pd.Timestamp(result.end_ts, unit="ns") == pd.Timestamp.max
+
+
+def _random_integers(size, dtype):
+    # We do not generate integers outside the int64 range
+    platform_int_info = np.iinfo("int_")
+    iinfo = np.iinfo(dtype)
+    return np.random.randint(
+        max(iinfo.min, platform_int_info.min), min(iinfo.max, platform_int_info.max), size=size
+    ).astype(dtype)
+
+
+def test_read_unsorted_date_range_dataframe(lmdb_version_store):
+    symbol = "read_unsorted_incr"
+
+    num_initial_rows = 20
+    initial_timestamp = pd.Timestamp("2019-01-01")
+    dtidx = pd.date_range(initial_timestamp, periods=num_initial_rows)
+
+    df = pd.DataFrame(
+        {
+            "uint8": _random_integers(num_initial_rows, np.uint8),
+            "uint32": _random_integers(num_initial_rows, np.uint32),
+        },
+        index=np.roll(dtidx, 3),
+    )
+
+    assert df.index.is_monotonic_increasing == False
+    assert df.index.is_monotonic_decreasing == False
+    lmdb_version_store.write(symbol, df)
+    info = lmdb_version_store.get_info(symbol)
+    assert info["sorted"] == "UNSORTED"
+    with pytest.raises(InternalException) as e_info:
```

Comment:
`InternalException` is a bit generic.

For all new errors, a unique error code should be assigned and asserted in tests.